cdap-distributions/bin/build_artifact_upload.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+#
+# Copyright © 2012-2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Receive artifacts and upload them directly to docs1 (or docs2 if there is an issue?)
+# This script expects to receive 3 parameters in this order:
+#   $1 user                     REMOTE_USER
+#   $2 remote host              REMOTE_HOST
+#   $3 target directory         REMOTE_INCOMING_DIR
+###############################################################################
+#S3_DRY_RUN=--dry-run
+DEBUG="yes"
+
+# Vars
+RUN_DATE=`date '+%Y%m%d_%R'`
+MAINDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+SCRIPT=`basename ${BASH_SOURCE[0]}`                     ### Set Script Name variable
+LOG_DIR=${MAINDIR}/logs                                 ### log directory
+REMOTE_USER=${1}                                        ### remote user
+REMOTE_HOST=${2:-127.0.0.1}                             ### remote host
+REMOTE_INCOMING_DIR=${3}                                ### target directory on remote host
+REMOTE_BASE_DIR="${REMOTE_USER}@${REMOTE_HOST}:${REMOTE_INCOMING_DIR}"
+BUILD_RELEASE_DIRS=*/target                             ### Source directories
+BUILD_PACKAGE=${BUILD_PACKAGE:-cdap}
+
+# notifications and logging
+LOG_FILE=build_sync_logs_${RUN_DATE}.txt
+LOGS=$LOG_DIR/$LOG_FILE
+exec &> >(tee -a "${LOGS}")
```

Comment:
There's no real reason for logging. This is meant to be run from a CI system, which normally captures logs. In this case, we want it to output to STDOUT.